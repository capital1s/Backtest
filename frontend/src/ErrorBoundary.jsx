
import React from "react";
import PropTypes from "prop-types";

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false, error: null };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true, error };
  }

  // eslint-disable-next-line no-unused-vars
  componentDidCatch(error, errorInfo) {
    // You can log errorInfo to an error reporting service here
    // console.error(error, errorInfo);
  }

  handleReset = () => {
    this.setState({ hasError: false, error: null });
  };

  render() {
    if (this.state.hasError) {
      return (
        <div role="alert" style={{ color: "red", padding: "2rem" }}>
          <h2>Something went wrong.</h2>
          <pre>{this.state.error?.toString()}</pre>
          <button onClick={this.handleReset} aria-label="Reset error boundary">Try Again</button>
        </div>
      );
    }
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ErrorBoundary;
