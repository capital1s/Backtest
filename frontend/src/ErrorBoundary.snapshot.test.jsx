// @vitest-environment jsdom
import React from "react";
import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import ErrorBoundary from "./ErrorBoundary";

describe("ErrorBoundary", () => {
  it("matches snapshot with children", () => {
    const { asFragment } = render(
      <ErrorBoundary>
        <div>Safe Child</div>
      </ErrorBoundary>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with error fallback", () => {
    const ProblemChild = () => { throw new Error("Test error"); };
    const { asFragment } = render(
      <ErrorBoundary>
        <ProblemChild />
      </ErrorBoundary>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
