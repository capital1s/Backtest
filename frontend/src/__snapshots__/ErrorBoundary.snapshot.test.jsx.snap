// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBoundary > matches snapshot with children 1`] = `
<DocumentFragment>
  <div>
    Safe Child
  </div>
</DocumentFragment>
`;

exports[`ErrorBoundary > matches snapshot with error fallback 1`] = `
<DocumentFragment>
  <div
    role="alert"
    style="color: red; padding: 2rem;"
  >
    <h2>
      Something went wrong.
    </h2>
    <pre>
      Error: Test error
    </pre>
    <button
      aria-label="Reset error boundary"
    >
      Try Again
    </button>
  </div>
</DocumentFragment>
`;
